### PR TITLE
Emit moveToTrash event only for the deleting user

### PR DESCRIPTION
**Pull request by juliusknorr**
This is the same issue as in https://github.com/nextcloud/server/pull/16624 which causes a huge loading time caused by iterating over all mounts for a file id for all users if getById is called on the RootFolder:

https://github.com/nextcloud/server/blob/e43b341b04c5e13424f2d833803c0e444f8ab3b2/lib/private/Files/Node/Folder.php#L292

Since the moveToTrash event is only used in e2ee it should be fine to just check for the file in the users mount who is trying to delete the file.

### CODE REVIEW

**Comment by blizzz** [comment]
don't we need the user exists check anymore when requesting the user folder instance?

**Comment by juliushaertl** [comment]
Hm I currently fail to understand why that unit test is failing with that change :/

**Comment by rullzer** [comment]
ok so the issue with the failing test is that the users storage gets initisialized thus for the final change of the lock the user mount is there. Causing the lock to be different.

So assume 'testUser'

* Then the first read lock is on '/' with path 'testUser/files/test.txt'
* Second change is on '/' with 'testUser/files/test.txt'
* Then the userfolder gets initialized
* Thirs lock change is on '/testUser' with path 'files/test.txt'

The last one fails because it wants to release the exclusive lock (which isn't there)

**Review by skjnldsv — APPROVED** [review]

File: apps/files_trashbin/lib/Storage.php [hunk 1]
```diff
@@ -125,10 +125,21 @@ public function rmdir($path) {
...
+		$normalized = Filesystem::normalizePath($this->mountPoint . '/' . $path);
+		$parts = explode('/', $normalized);
+		if (count($parts) < 4) {
+			return false;
+		}
 
 		// check if there is a app which want to disable the trash bin for this file
 		$fileId = $this->storage->getCache()->getId($path);
-		$nodes = $this->rootFolder->getById($fileId);
+		$owner = $this->storage->getOwner($path);
```
Comment (juliushaertl) [inline apps/files_trashbin/lib/Storage.php]: @rullzer Are there actually cases where the trash bin storage doesn't have an owner?

Comment (rullzer) [reply apps/files_trashbin/lib/Storage.php]: Well it is an issue with the storage stuff.
If the storage is not setup for the user. It will be null

Comment (juliushaertl) [reply apps/files_trashbin/lib/Storage.php]: Ah right, that should not be an issue then for the code paths where the slow down was experienced then.

**Review by blizzz — APPROVED** [review]

**Pull request merged**